### ISSUE TITLE
hcloud-cloud-controller-manager is not updating external ip although routing works normaly

### ISSUE THREAD

**Issue by NikolaosEfthymiou** [hetznercloud/hcloud-cloud-controller-manager#811]
### TL;DR

Hello, i fell into a weird issue, my loadbalancer works as expected in this scenario and i can see my nginx website visiting the IP of loadbalancer that i found on hcloud console
however
hcloud-cloud-controller-manager is not updating external ip of the service, instead its stuck on pending although everything works as it should. any ideas?

```kubectl get services
NAME                   TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)        AGE
custom-nginx-service   LoadBalancer   10.43.31.21    <pending>     80:30594/TCP   15m
```

```
Name:                     custom-nginx-service
Namespace:                default
Labels:                   <none>
Annotations:              load-balancer.hetzner.cloud/disable-public-network: true
                          load-balancer.hetzner.cloud/location: nbg1
                          load-balancer.hetzner.cloud/use-private-ip: true
Selector:                 app=custom-nginx
Type:                     LoadBalancer
IP Family Policy:         SingleStack
IP Families:              IPv4
IP:                       10.43.31.21
IPs:                      10.43.31.21
Port:                     <unset>  80/TCP
TargetPort:               80/TCP
NodePort:                 <unset>  30594/TCP
Endpoints:                10.42.2.17:80
Session Affinity:         None
External Traffic Policy:  Cluster
Events:
  Type    Reason                Age   From                Message
  ----    ------                ----  ----                -------
  Normal  EnsuringLoadBalancer  16m   service-controller  Ensuring load balancer
  Normal  EnsuredLoadBalancer   15m   service-controller  Ensured load balancer
```

### Expected behavior

external ip is updated on the service description

### Observed behavior

external ip is not updated on the service description

### Minimal working example

```
apiVersion: v1
kind: Service
metadata:
  name: custom-nginx-service
  namespace: default
  annotations:
    load-balancer.hetzner.cloud/location: "nbg1"
    load-balancer.hetzner.cloud/use-private-ip: "true"
    load-balancer.hetzner.cloud/disable-public-network: "true"
spec:
  selector:
    app: custom-nginx
  ports:
  - protocol: TCP
    port: 80
    targetPort: 80
  type: LoadBalancer
```

### Log output

```plain text
I1130 18:09:24.288524 1 load_balancers.go:192] "reload HC Load Balancer" op="hcloud/loadBalancers.EnsureLoadBalancer" loadBalancerID=2195035
I1130 18:09:24.623103 1 event.go:389] "Event occurred" object="default/custom-nginx-service" fieldPath="" kind="Service" apiVersion="v1" type="Normal" reason="EnsuredLoadBalancer" message="Ensured load balancer"
```

### Additional information

kubernetes cluster was created using kube-hetzner terraform module. i disabled the creation of loadbalancer via the module and i let hcloud-cloud-controller-manager to create its own which properly does

**Comment by NikolaosEfthymiou** [comment]
i found a similar issue, https://github.com/hetznercloud/hcloud-cloud-controller-manager/issues/50 , its probably the same case, will close as duplicate if its resolved via this

**Comment by NikolaosEfthymiou** [comment]
A little bit of extra info, if im not using those annotations, the external ip is normaly updated with the correct public ip. So the issue is its not updating the private external ip when configured to use private
```
    load-balancer.hetzner.cloud/use-private-ip: "true"
    load-balancer.hetzner.cloud/disable-public-network: "true"
```

**Comment by lukasmetzner** [comment]
Hey,

I can't reproduce your issue. Could you maybe give me more information about your configuration of the hcloud-cloud-controller-manager via `kubectl -n kube-system describe deployments.apps hcloud-cloud-controller-manager`.

I am also not sure if I understood correctly what your goal is.

**Comment by NikolaosEfthymiou** [comment]
> I am also not sure if I understood correctly what your goal is. Could you give me some more details :D

Okay that's a minimal example:
```
apiVersion: v1
kind: Pod
metadata:
  name: nginx-pod
  labels:
    app: nginx
spec:
  containers:
  - name: nginx
    image: nginx:latest
    ports:
    - containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: nginx-service
  annotations:
    load-balancer.hetzner.cloud/location: "nbg1"
    load-balancer.hetzner.cloud/use-private-ip: "true"
    load-balancer.hetzner.cloud/disable-public-network: "true"

spec:
  type: LoadBalancer
  selector:
    app: nginx
  ports:
  - protocol: TCP
    port: 80
    targetPort: 80
```

this creates a loadbalancer
![image](https://github.com/user-attachments/assets/292fab91-15ad-46a5-863a-56b0ff2e5ed0)

if i visit the private ip of my loadbalancer i can normaly see my nginx as expected

![image](https://github.com/user-attachments/assets/863d09d0-6744-4d45-97b1-152f83057466)

The events on service seem fine as well
![image](https://github.com/user-attachments/assets/1e437a9b-40e9-465d-95c7-1a1d49323026)

However the service is not really updated with the loadbalancer EXTERNAL-IP and its stuck on pending somehow 
![image](https://github.com/user-attachments/assets/8e31f867-104e-4792-b43a-134751c7ffd1)

This scenario works perfectly when using public ip instead.. means the the public ip of loadbalancer is normaly updated on EXTERNAL-IP field of the service

kubectl -n kube-system describe deployments.apps hcloud-cloud-controller-manager

```
Name:                   hcloud-cloud-controller-manager
Namespace:              kube-system
CreationTimestamp:      Sun, 01 Dec 2024 17:34:00 +0000
Labels:                 <none>
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               app=hcloud-cloud-controller-manager
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           app=hcloud-cloud-controller-manager
  Service Account:  hcloud-cloud-controller-manager
  Containers:
   hcloud-cloud-controller-manager:
    Image:      docker.io/hetznercloud/hcloud-cloud-controller-manager:v1.21.0
    Port:       8233/TCP
    Host Port:  0/TCP
    Args:
      --cloud-provider=hcloud
      --leader-elect=false
      --allow-untagged-cloud
      --route-reconciliation-period=30s
      --allocate-node-cidrs=true
      --cluster-cidr=10.42.0.0/16
      --webhook-secure-port=0
    Requests:
      cpu:     100m
      memory:  50Mi
    Environment:
      HCLOUD_LOAD_BALANCERS_LOCATION:                 fsn1
      HCLOUD_LOAD_BALANCERS_USE_PRIVATE_IP:           true
      HCLOUD_LOAD_BALANCERS_ENABLED:                  true
      HCLOUD_LOAD_BALANCERS_DISABLE_PRIVATE_INGRESS:  true
      HCLOUD_TOKEN:                                   <set to the key 'token' in secret 'hcloud'>           Optional: false
      ROBOT_PASSWORD:                                 <set to the key 'robot-password' in secret 'hcloud'>  Optional: true
      ROBOT_USER:                                     <set to the key 'robot-user' in secret 'hcloud'>      Optional: true
      HCLOUD_NETWORK:                                 <set to the key 'network' in secret 'hcloud'>         Optional: false
    Mounts:                                           <none>
  Volumes:                                            <none>
  Priority Class Name:                                system-cluster-critical
  Node-Selectors:                                     <none>
  Tolerations:                                        CriticalAddonsOnly op=Exists
                                                      node-role.kubernetes.io/control-plane:NoSchedule op=Exists
                                                      node-role.kubernetes.io/master:NoSchedule op=Exists
                                                      node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule
                                                      node.kubernetes.io/not-ready:NoExecute
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      True    MinimumReplicasAvailable
OldReplicaSets:  <none>
NewReplicaSet:   hcloud-cloud-controller-manager-5f6dbbcd9 (1/1 replicas created)
Events:          <none>
```
Finally i just figured out that there is a variable HCLOUD_LOAD_BALANCERS_DISABLE_PRIVATE_INGRESS:  true on
deployments.apps hcloud-cloud-controller-manager which if disabled it updates the service normaly
![image](https://github.com/user-attachments/assets/b65745b2-775b-4272-9953-a5dcbcdc9926)

So if HCLOUD_LOAD_BALANCERS_DISABLE_PRIVATE_INGRESS = true although its supposed to disable the networking, it still allows it, its just not updating the EXTERNAL-IP field hmmm

Anyway thanks for the comment really helped me to understand my issue
